plugins/source/aws/resources/services/directconnect/gateways_fetch.go

```diff
@@ -66,8 +68,15 @@ func fetchDirectconnectGatewayAttachments(ctx context.Context, meta schema.Clien
 	return nil
 }
 
-func resolveGatewayARN() schema.ColumnResolver {
-	return client.ResolveARNWithAccount(client.DirectConnectService, func(resource *schema.Resource) ([]string, error) {
-		return []string{"dx-gateway", *resource.Item.(types.DirectConnectGateway).DirectConnectGatewayId}, nil
-	})
+func resolveGatewayARN(_ context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	cl := meta.(*client.Client)
+	gw := resource.Item.(types.DirectConnectGateway)
+
+	return resource.Set(c.Name, arn.ARN{
+		Partition: cl.Partition,
+		Service:   "directconnect",
+		AccountID: *gw.OwnerAccount,
+		Region:    cl.Region,
```

Comment:
`cl.Region` here might be wrong maybe just add `Region` as one of the PKs and leave this empty.

Reply:
https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsdirectconnect.html#awsdirectconnect-resources-for-iam-policies
`dx-gateway` doesn't have the Region in ARN, so, I think, it'll be wrong to add it here.

Reply:
Good doc, fixed